### ISSUE TITLE
LinesCounter

### ISSUE THREAD

**Issue by sfactor** [khaasnew/nsudotnet#1]
- старые проекты удалять не надо
- `line.Replace("\t", string.Empty)` делать не обязательно — `string.IsNullOrWhiteSpace(line)` считает табуляции за whitespace
- алгоритм анализа строк очень неэффективный: создается очень много временных объектов. Всё то же самое можно было сделать за один проход по строке

Пока могу поставить 1 балл

**Comment by khaasnew** [comment]
Из строки убрала подстроку с комментарием(в 3-х местах). При этом сделалась новая измененная строка, и она проверяется на наличие значимых символов.Снйчас подстроки не делаются, а просто при проходе по строке выполняется проверка на значимость символа. Значимый - больше проверять не надо, учитываем эту строку и break.

**Comment by sfactor** [comment]
Так, наведи порядок с папками. Сейчас я вижу три LineCounter-а — в какой мне смотреть?

**Comment by khaasnew** [comment]
2 папки. Тот, что lines_counter_new, с последними исправлениями.

**Comment by sfactor** [comment]
Вот эти два прохода по строке совершенно лишние:

```
var singleCommentStart = line.IndexOf(singleStart);
var multiCommentStart = line.IndexOf(multiStart);
```

(напоминаю, что string.IndexOf() — это проход по строке с линейной сложностью)

Ну и нахождение подстроки осталось, без него тоже вполне можно обойтись:

```
var notComment = line.Substring(endCommentPosition + multiEnd.Length);
```

Еще раз: весь алгоритм делается за один проход по строке (обычным for-ом, без поисков подстроки и прочего)

**Comment by khaasnew** [comment]
исправлено.

**Comment by sfactor** [comment]
Неа, тут совсем не один проход по строке.

`string.Where()` тебя подвел:) Ну и поиск подстроки один остался.
